components/admin/permit-holders/PermitHolderTypeahead.tsx

```diff
@@ -0,0 +1,73 @@
+import { useQuery } from '@apollo/client'; // Apollo
+import { Text } from '@chakra-ui/layout'; //Chakra UI
+import Typeahead from '@components/Typeahead'; // Typeahead component
+import { formatDateYYYYMMDD } from '@lib/utils/format'; // Date formatter util
+import {
+  GetPermitHoldersRequest,
+  GetPermitHoldersResponse,
+  GET_PERMIT_HOLDERS_QUERY,
+  PermitHolder,
+} from '@tools/pages/admin/permit-holders/get-permit-holders'; // Permit holders GQL query
+import { useState } from 'react'; // React
+
+// Permit Holder Typeahead props
+type Props = {
+  setSelected: (selected: PermitHolder | undefined) => void; // record selected permit holder
+};
+
+/**
+ * Typeahead component to search for permit holders via asyncronous queries
+ * @returns Permit holder typeahead component to search for permit holders
+ */
+export default function PermitHolderTypeahead(props: Props) {
+  const { setSelected } = props;
+  const [isTypeaheadLoading, setIsTypeaheadLoading] = useState(false);
+  const [permitHolderResults, setPermitHolderResults] = useState<PermitHolder[]>([]);
+  const [searchString, setSearchString] = useState<string>();
+
+  // permit holders query
+  useQuery<GetPermitHoldersResponse, GetPermitHoldersRequest>(GET_PERMIT_HOLDERS_QUERY, {
+    variables: {
+      filter: {
+        search: searchString,
+      },
+    },
+    onCompleted: ({ applicants: { result } }) => {
+      setPermitHolderResults(
+        result.map(record => ({
+          ...record,
+        }))
```

Comment:
I don't think we need to make a deep copy here, but I could be wrong

Reply:
I think we do have to since the `result` which is returned is readonly, so vscode complains about it if we don't deep copy.

Reply:
Got it